runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -10022,15 +10036,24 @@ VMinlineCompareAndSwap(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mn
    if (TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads() && isObj)
       {
       TR::Register* tempReadBarrier = cg->allocateRegister();
+      if (TR::Compiler->om.shouldReplaceGuardedLoadWithSoftwareReadBarrier())
+         {
+         // Software concurrent scavenge
+         bool shouldDecompressField = (casOp == TR::InstOpCode::CS) ? true : false;
```

Comment:
How about just:

```
const bool shouldDecompressField = (casOp == TR::InstOpCode::CS);
```

Or better yet just fold that directly into the call to `generateSoftwareReadBarrier`. Also the comment there is not useful since the `if` statement above describes exactly what is going on.

Reply:
The flag is folded into the call.

Reply:
We should clean up the comments as well.

Reply:
Removed unneeded comments in 911a46b009